symbols/overlay29.yml

```
@@ -412,13 +473,45 @@ overlay29:
       address:
         NA: 0x22EB178
         EU: 0x22EBB28
+        JP: 0x22EC7E0
       description: |-
         The user entity attempts to switch places with the target entity (i.e. by the effect of the Switcher Orb). 
         
         The function checks for the Suction Cups ability for both the user and the target, and for the Mold Breaker ability on the user.
         
         r0: pointer to user entity
         r1: pointer to target entity
+    - name: FloorLoop
+      address:
+        NA: 0x22EBD08
+        EU: 0x22EC6B8
+        JP: 0x22ED370
+      description: |-
+        The main function which updates the state of the floor. Called in a loop by InitializeDungeon while FloorIsOver returns false.
+        
+        The parameter is always equal to 1 and is used as parameter for another unknown function.
+        
+        r0: int
+    - name: TrySpawnMonsterAndActivatePlusMinus
+      address:
+        NA: 0x22EC6DC
+        EU: 0x22ED08C
+        JP: 0x22EDD44
+      description: |-
+        Called at the beginning of FloorLoop. Executed only if FRACTIONAL_TURN_SEQUENCE[fractional_turn * 2] is not 0.
+        
+        First it calls TrySpawnMonsterAndTickSpawnCounter, then tries to activate the Plus and Minus abilities for both allies and enemies, and finally calls TryForcedLoss.
+        
+        No params.
+    - name: FloorIsOver
+      address:
+        NA: 0x22EC7E8
+        EU: 0x22ED198
+        JP: 0x22EDE50
+      description: |-
+        Returns true if the player will advance to the next floor on the next turn, and in that case sets the floor_loop_status in the dungeon struct to 2.
+        
+        return: bool
```

```suggestion
        Checks if the current floor should end, and updates dungeon::floor_loop_status if required.
        If the player has been defeated, sets dungeon::floor_loop_status to floor_loop_status::FLOOR_LOOP_LEADER_FAINTED.
        If dungeon::end_floor_flag is 1 or 2, sets sets dungeon::floor_loop_status to floor_loop_status::FLOOR_LOOP_NEXT_FLOOR.
        
        return: True if the current floor should end
```
Looking at my notes to write this made me realize that dungeon::end_floor_flag is actually an `uint8_t`, not a `bool`. We'll have to change that at some point.

---

What do the different values mean? If we're updating the name, we should update the field's comment too.

---

I noticed that `quicksave_flag`'s comment is also outdated. would changing it to this be ok?
```C
// 0x6: If equal to 1 or 2, the floor will be advanced at the end of the turn, 
// unless the leader fainted.
uint8_t end_floor_flag;
// 0x7: If set in tandem with either end_floor_flag or end_floor_no_death_check_flag, 
// a quicksave will occur instead of a floor advance.
bool quicksave_flag;
```

---

Yeah that sounds good.